### PR TITLE
Boost: Add edge-cache header on WP Cloud

### PR DESCRIPTION
**Pull request by haqadn**
## Proposed changes:
* If it's a WP Cloud site, add a header that helps cache the minified file.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pfFwAW-nW-p2#comment-245

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Install it on a WPCOM site.
* Enable minify JS/CSS.
* Check the header on `_jb_static` files that loads.
* `A8c-Edge-Cache: cache` header should be present.

### CODE REVIEW

File: projects/plugins/boost/app/lib/minify/functions-service.php [hunk 1]
```diff
@@ -60,6 +60,9 @@ function jetpack_boost_page_optimize_service_request() {
 			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 			$etag = '"' . md5( file_get_contents( $cache_file ) ) . '"';
 
+			if ( defined( 'IS_ATOMIC' ) && IS_ATOMIC ) {
```
Comment (thingalon) [inline projects/plugins/boost/app/lib/minify/functions-service.php]: I don't think `IS_ATOMIC` is defined on Pressable sites FYI.

**Comment by donnchawp** [comment]
I tried it on my test wpcom site with upgrades and I didn't see the header, but it appears to be because `Config::can_use_cache();` returned false. Investigating..

**Comment by haqadn** [comment]
I have instead used the ATOMIC_CLIENT_ID constant, which is available on both Atomic and Pressable. However, I couldn't see the header even when I removed the condition and served the header all the time in atomic. Is the header filtered out by the edge? @thingalon

I also noticed that the files aren't cached in wpcom, which is a bummer. I think this will get resolved when we start serving from static files. boost-cache directory seems to be doing just fine. @thingalon previously recommended serving from inside the uploads folder. Do you see any downsides to serving from `wp-content/boost-cache`?

**Comment by donnchawp** [comment]
I think the `boost-cache` directory is probably a good place to put these files. We can be in complete control of them, and we'll be sure we're not interfering with files from either the blog or another plugin.
If we rely on the 404 handler to regenerate static files then `wp-content/boost-cache/` and a sub directory like `static` will be easy to identify.
However, if we do this, then we have to be sure that the `boost-cache` directory itself isn't removed if the "Page Cache" module is toggled off.

**Comment by thingalon** [comment]
Ah, sorry I missed this last week @haqadn. :)

> I have instead used the ATOMIC_CLIENT_ID constant, which is available on both Atomic and Pressable. However, I couldn't see the header even when I removed the condition and served the header all the time in atomic. Is the header filtered out by the edge? @thingalon

Yes, we strip that header out on its way through the Edge. However, you can check if a given request was cached or not by looking at the `x-ac` header in the response.

If you request the same URL a few times and see `HIT` in there, you know that you've successfully hit the cache. :) Make sure you're not authenticated, as a login cookie causes a cache bypass. Using `curl` should do it.

> I also noticed that the files aren't cached in wpcom, which is a bummer. I think this will get resolved when we start serving from static files. boost-cache directory seems to be doing just fine. @thingalon previously recommended serving from inside the uploads folder. Do you see any downsides to serving from `wp-content/boost-cache`?

I believe that `wp-content/boost-cache` should be fine.

I'm not sure which files aren't being cached on wpcom; if you can send me details, I can help out with that.

**Review by dilirity — APPROVED** [review]
This works on Atomic and Pressable.

:shipit:

**Comment by thingalon** [comment]
@Automattic/heart-of-gold - can y'all do me a favor and give me a ping when this goes out in a release so I can test it? :)

**Comment by haqadn** [comment]
Hey @thingalon! The release is here: https://github.com/Automattic/jetpack-boost-production/releases/tag/3.7.0

We didn't update the stable tag yet(as of now) so you'll have to download it from the gh release.

**Comment by thingalon** [comment]
Tested 3.7.0, and it does indeed fix the issue. Thank you! :)

**Pull request merged**